hypyp/prep.py

```diff
@@ -66,68 +76,71 @@ def ICA_choice_comp(icas: list, epochs: list) -> list:
 
     # choosing participant and its component as a template for the other participant
     # if do not want to apply ICA on the data, do not fill the answer
-    subj_numb = input("Which participant ICA do you want"
-                      " to use as a template for artifact rejection?"
-                      " Index begins at zero. (If you do not want to apply"
-                      " ICA on your data, do not enter nothing and press enter.)")
-    comp_number = input("Which IC do you want to use as a template?"
+    subject_id = input("Which participant ICA do you want"
+                        " to use as a template for artifact rejection?"
+                        " Index begins at zero. (If you do not want to apply"
+                        " ICA on your data, do not enter nothing and press enter.)")
+
+    component_id = input("Which IC do you want to use as a template?"
                         " Index begins at zero. (If you did not choose"
-                        "a participant number at first question,"
-                        "then do not enter nothing and press enter again"
-                        "to not apply ICA on your data)")
-
-    # applying ICA
-    if (len(subj_numb) != 0 and len(comp_number) != 0):
-        cleaned_epochs_ICA = ICA_apply(icas,
-                                       int(subj_numb),
-                                       int(comp_number),
-                                       epochs)
-    else:
-        cleaned_epochs_ICA = epochs
+                        " a participant number at first question,"
+                        " then do not enter nothing and press enter again"
+                        " to not apply ICA on your data)")
 
-    return cleaned_epochs_ICA
+    if (len(subject_id) == 0 or len(component_id) == 0):
+        return epochs
+
+    return ICA_apply(icas, int(subject_id), int(component_id), epochs)
 
 
-def ICA_apply(icas: int, subj_number: int, comp_number: int, epochs: list) -> list:
+def ICA_apply(icas: List[ICA], subject_id: int, component_id: int, epochs: List[mne.Epochs], plot: bool = True) -> List[mne.Epochs]:
     """
     Applies ICA with template model from 1 participant in the dyad.
     See ICA_choice_comp for a detailed description of the parameters and output.
     """
 
-    cleaned_epochs_ICA = []
+    cleaned_epochs_ICA: List[ICA] = []
+
     # selecting which ICs corresponding to the template
-    template_eog_component = icas[subj_number].get_components()[:, comp_number]
+    template_eog_component = icas[subject_id].get_components()[:, component_id]
 
     # applying corrmap with at least 1 component detected for each subj
-    fig_template, fig_detected = corrmap(icas,
-                                         template=template_eog_component,
-                                         threshold=0.9,
-                                         label='blink',
-                                         ch_type='eeg')
+    corrmap(icas,
```

Comment:
Added `plot` as `ICA_apply` argument (to avoid plots launching GUI windows during tests). The presence/absence of `plot` argument here change the return type of `corrmap`, so we cannot always unpack `fig_template` and `fig_detected`. Anyway, they are not used.